homeassistant/components/sonos/speaker.py

```diff
@@ -857,21 +854,33 @@ def _get_soco_group() -> list[str]:
 
         async def _async_extract_group(event: SonosEvent | None) -> list[str]:
             """Extract group layout from a topology event."""
-            group = event and event.zone_player_uui_ds_in_group
-            if group:
-                assert isinstance(group, str)
-                return group.split(",")
-
+            try:
+                group = event and event.zone_player_uui_ds_in_group
+                if group:
+                    assert isinstance(group, str)
+                    return group.split(",")
+            except AttributeError:
+                pass
```

Comment:
Is the `try` to protect from `event.zone_player_uui_ds_in_group` not being set? If that's the case, would something like this be simpler?
```suggestion
            if group := (event and getattr(event, "zone_player_uui_ds_in_group")):
                assert isinstance(group, str)
                return group.split(",")
```

Reply:
Yes, that is much simpler and gets rid of 4 lines of code.  Thank you!